backends/concrete-cpu/implementation/src/c_api/fheint.rs

```diff
@@ -0,0 +1,180 @@
+use super::utils::nounwind;
+use core::slice;
+use std::io::Cursor;
+use tfhe::core_crypto::prelude::*;
+use tfhe::integer::ciphertext::Expandable;
+use tfhe::integer::IntegerCiphertext;
+use tfhe::shortint::parameters::{Degree, NoiseLevel};
+use tfhe::shortint::{CarryModulus, Ciphertext, MessageModulus};
+use tfhe::FheUint8;
+
+#[repr(C)]
+pub struct TfhersFheIntDescription {
+    width: usize,
+    is_signed: bool,
+    lwe_size: usize,
+    n_cts: usize,
+    degree: usize,
+    noise_level: usize,
+    message_modulus: usize,
+    carry_modulus: usize,
+    ks_first: bool,
+}
+
+impl TfhersFheIntDescription {
+    fn zero() -> TfhersFheIntDescription {
+        TfhersFheIntDescription {
+            width: 0,
+            is_signed: false,
+            lwe_size: 0,
+            n_cts: 0,
+            degree: 0,
+            noise_level: 0,
+            message_modulus: 0,
+            carry_modulus: 0,
+            ks_first: false,
+        }
+    }
+
+    /// Create a `Ciphertext` from the lwe and `self` metadata
+    fn ct_from_lwe(&self, lwe: LweCiphertext<Vec<u64>>) -> Ciphertext {
+        Ciphertext::new(
+            lwe,
+            // TODO: how to set degree here?
+            Degree::new(self.degree),
+            // TODO: how to set noise here?
```

Comment:
UNKNOWN should be safe and trigger a PBS on our end

Reply:
to be clear it still means that the noise in the ciphertext must be in the valid interval/what the TFHE-rs parameters were optimized for, otherwise we can get garbage outputs

Reply:
for the degree it is the max possible value contained in that ciphertext, so for 2_2 parameters (2 bits of message and 2 bits of carry) the degree at encryption time is 3 as we could have encrypted 0, 1, 2 or 3 (unless we do packing) in which case we can use also the carry bits which means the max degree is 15 (4 bits set to 1)

if the value is bigger then the padding bit must have been trashed which means we can't guarantee correctness

Reply:
as we don't do any operations on those, and we only use the msg bits in the output, I guess we can always set those to `2**msg_bits - 1`

Reply:
yep if you "just" store stuff within the message modulus space then the degree is what you wrote